File: mascota/src/Application/casos_de_uso/CreatePetUseCase.ts
import { IPetRepository } from '../../Domain/Repositories/IPetRepository';
import { Pet } from '../../Domain/entities/Pet';

export class CreatePetUseCase {
    constructor(private repository: IPetRepository) {}

    async execute(pet: Pet): Promise<void> {
        await this.repository.create(pet);
    }
}

File: mascotapy/src/Mascota/Application/Casos_de_uso/CreatePetUseCase.py
from Mascota.Domain.Entities.Pet import Pet  
from Mascota.Domain.Repositories.IPetRepository import IPetRepository 

class CreatePetUseCase:
    def __init__(self, repository: IPetRepository):
        self.repository = repository

    def execute(self, pet: Pet) -> None:
        """Ejecuta la creación de una nueva mascota."""
        self.repository.create(pet)
        print(f"✔️ Mascota creada: {pet}")
